openid-sharedsignals-framework-1_0.md

```diff
@@ -397,14 +397,107 @@ below MAY define certain members within a Complex Subject to be Critical. A SSF
 Receiver MUST discard any event that contains a Subject with a Critical member
 that it is unable to process.
 
-# Event Properties  {#properties}
+# Events  {#events}
 
-Additional members about an event may be included in the "events" claim. Some
-of these members are required and specified as such in the respective event
-types specs. If a Transmitter determines that it needs to include additional
-members that are not specified in the event types spec, then the name of such
-members MUST be a URI. The discoverability of all additional members is
-specified in the Discovery section ({{discovery}}).
+## Security Event Token Profile {#set-profle}
+The Shared Signals Framework profiles the Security Event Token (SET)
+{{RFC8417}} specification by defining certain properties of SETs as described in this section.
+
+### Explicit Typing of SETs {#explicit-typing}
+SSF events MUST use explicit typing as defined in Section 2.3 of {{RFC8417}}.
+
+~~~ json
+{
+  "typ":"secevent+jwt",
+  "alg":"HS256"
+}
+~~~
+{: title="Explicitly Typed JOSE Header" #explicit-type-header}
+
+The purpose is defense against confusion with other JWTs, as described in
+Sections 4.5, 4.6 and 4.7 of {{RFC8417}}. While current Id Token {{OpenID.Core}}
+validators may not be using the "typ" header parameter, requiring it for SSF
+SETs guarantees a distinct value for future validators.
+
+### SSF Event Subject {#event-subjects}
+The primary Subject Member of SSF events is described in the "Subject Members" section ({{subject-ids}}). The JWT "sub" claim MUST NOT be present in any SET containing
+an SSF event.
+
+### Distinguishing SETs from other Kinds of JWTs
+Of particular concern is the possibility that SETs are confused for other kinds
+of JWTs. The Security Considerations section of {{RFC8417}} has several sub-sections
+on this subject. The Shared Signals Framework requires further restrictions:
+
+* The "sub" claim MUST NOT be present, as described in {{event-subjects}}.
+* SSF SETs MUST use explicit typing, as described in {{explicit-typing}}.
+* The "exp" claim MUST NOT be present, as described in {{exp-claim}}.
+
+### Signature Key Resolution {#signature-key-resolution}
+The signature key can be obtained through "jwks_uri", see {{discovery}}.
+
+### The "iss" Claim {#iss-claim}
+The "iss" claim MUST match the "iss" value in the Stream Configuration data for the stream
+that the event is sent on. Receivers MUST validate that this claim matches the "iss"
+in the Stream Configuration data, as well as the Issuer from which the Receiver requested
+the Transmitter Configuration data.
+
+### The "exp" Claim {#exp-claim}
+The "exp" claim MUST NOT be used in SSF SETs.
```

Comment:
Should we remove "SSF SETs"? Isn't the same true for CAEP and RISC?

Reply:
Updated, PTAL